recipe/meta.yaml

```diff
@@ -1,55 +1,305 @@
 {% set name = "google-api-core" %}
-{% set version = "1.8.2" %}
-{% set sha256 = "16b14c0492154110d869a08be22d27440010057fd826205303c7edaa1c3cd71e" %}
+{% set version = "1.9.0" %}
+{% set sha256 = "fc1fea74bd863fb71486066e0c6b3a4dad26fb70ec61a0edcada8637feb77c68" %}
 {% set build = 0 %}
 
 package:
-  name: {{ name|lower }}
+  name: {{ name|lower }}-feedstock
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
   url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
   sha256: {{ sha256 }}
 
 build:
-  number: 0
-  script: python -m pip install --no-deps --ignore-installed .
-
-requirements:
-  build:
-    - python
-    - setuptools
-    - pip
-
-  run:
-    - python
-    - googleapis-common-protos >=1.5.3,<2.0dev
-    - protobuf >=3.4.0
-    - google-auth >=0.4.0,<2.0.0dev
-    - requests >=2.18.0,<3.0.0dev
-    - setuptools >=34.0.0
-    - six >=1.10.0
-    - pytz
-    # Backport of concurrent.futures for Python < 3.2.
-    - futures >=3.0.0  # [py2k]
-
-test:
-  imports:
-    - google.api_core.exceptions
+  number: {{ build }}
+
+outputs:
+  - name: {{ name }}
+    build:
+      script: python -m pip install . --no-deps --ignore-installed --verbose
+    requirements:
+      host:
+        - python
+        - pip >=18.1
+        - futures >=3.2.0  # [py<32]
+        - google-auth >=0.4.0
+        - googleapis-common-protos >=1.5.3
+        - pytz >=2019.1
+        - requests >=2.18.0
+        - setuptools >=34.0.0
+        - six >=1.10.0
+      run:
+        - {{ pin_compatible('futures') }}  # [py<32]
+        - {{ pin_compatible('google-auth') }}
+        - {{ pin_compatible('googleapis-common-protos', min_pin='x', max_pin='x') }}
+        - {{ pin_compatible('pytz') }}
+        - {{ pin_compatible('requests', min_pin='x', max_pin='x') }}
```

Comment:
Why change from the default for `requests` and `googleapis-common-protos`?

Reply:
By default are you referring to the version upper bound `<2.0dev`?

The thing that prompted me to remove the upper bound was I noticed in the conda build logs version requirements like [ >= 1.5.3, != 1.5.4, < 2.0dev](https://github.com/googleapis/google-cloud-python/blob/master/api_core/setup.py#L32) from setup.py raise an invalid package error.

2.0dev release doesn't exists in the conda-forge organization or on pypi so it's a bit confusing.

Pull request updated, default upper bound set.

Reply:
By "default", I mean you're explicitly setting `min_pin='x'`, whereas the default is `min_pin='x.x.x.x.x'` or something like that.

> `2.0dev` release doesn't exists in the conda-forge organization or on pypi so it's a bit confusing.

That's intentional. It's to lock into the 1.x packages. 2.0dev is the earliest release that'll be on the 2.x version. `2.0dev < 2.0`.

Reply:
Ah I see, thanks.  I defined the pinning to be a bit more restrictive than the host version requirements for futures, pytz, setuptools and six using the default `package_name >=(current version),<(next major version)` to pin the major version.  

For google-auth, googleapis-common-protos and requests I defined the pinning to match the host version requirements.  We can remove min_pin if you prefer (use default) and only set max_pin or remove both min_pin and max_pin and use the default for both.  The condition on line [249](https://github.com/conda/conda-build/blob/master/conda_build/jinja_context.py#L249) ensures the min pinned version won't be below the current version.
```
if min_pin or lower_bound:
      compatibility = ">=" + str(version) + ","
```

Using the default is cleaner and less to maintain, PR updated.

To allow major semver changes you'd need to use pin_compatible with the lower_bound and upper_bound arguments.  Using pin_compatible with min_pin and max_pin the max allowable version change is within the current major version, same as the default.

`min_pin='x.x.x.x.x'` - pins the entire min version (major, minor, patch/bug and build), not sure what the 5 version part is or why you would need to use it.

[Pinning at the recipe level](https://github.com/conda/conda-build/blob/master/docs/source/resources/variants.rst#pinning-at-the-recipe-level) has a good description and examples including using lower_bound and upper_bound vs min_pin and max_pin.

[Build Variants](https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#id1):
> The number of version parts to pin is simply the number of things that are separated by .. For example, "x.x" pins major and minor version. "x" pins only major version.

[def pin_compatible(m, package_name, lower_bound=None, upper_bound=None, min_pin='x.x.x.x.x.x', max_pin='x', permit_undefined_jinja=False, exact=False, bypass_env_check=False):](https://github.com/conda/conda-build/blob/master/conda_build/jinja_context.py#L205)